### PR TITLE
Button: Rails documentation update

### PR DESCRIPTION
**Pull request by monicawheeler**
## Description
Simplify documentation for the Button component in Rails. 

### Goals of this PR
* Make the visual look and feel resemble the Pine component documentation
* Include more descriptions and update grammar where needed
* Update the props table with "deprecated" badges and add the missing information
* Include do and do not info blocks 
* Update the button link to React so it goes into the button component for React and does not trigger an error

## Screenshots

|  Before  |  After  |
|--------|--------|
| Screen Shot 2023-08-14 at 10 45 27 AM | Screen Shot 2023-08-14 at 10 45 17 AM |

## Testing in `sage-lib`
1. Navigate to http://localhost:4000/pages/component/button?tab=preview
2. Ensure the examples are complete
3. Navigate to http://localhost:4000/pages/component/button?tab=code
4. Ensure the code examples are easy to use and copyable.
5. Navigate to http://localhost:4000/pages/component/button?tab=props
6. Ensure descriptions are accurate and up to date
7. Navigate to http://localhost:4000/pages/component/button?tab=rules
8. Ensure the Do's and Don'ts are accurate and complete
9. Navigate to https://sage.kajabi.com/pages/component/button?tab=figma_embed
10. Ensure that you can view the button figma file 
11. Select the "React Component" link and ensure it navigates you to the button component in React.

## Testing in `kajabi-products`
1. (**LOW**) This work includes documentation changes only. No testing in KP is necessary.

### CODE REVIEW

**Review by ju-Skinner — APPROVED** [review]

**Review by QuintonJason — CHANGES_REQUESTED** [review]
These changes are great! My request is around the thought that reference to other components in the preview and don't and don't should link to them. What are your thoughts on this?

File: docs/app/views/examples/components/button/_rules_dont.html.erb [hunk 1]
```diff
@@ -1 +1,8 @@
-
+<ul>
+  <li>
+    <p>Directing users to a new page or different part within the same page. Instead, use a link.</p>
+  </li>
+  <li>
+    <p>When space is limited consider an icon only button.</p>
```
[the remark this answers is not in the document]
Comment (anechol) [reply docs/app/views/examples/components/button/_rules_dont.html.erb:6]: Agree here.

**Review by anechol — APPROVED** [review]
Agree with Quinton's comment about linking, but other than that, nice improvement!

**Review by pixelflips — APPROVED** [review]
Great update! Agree with others about linking but LGTM! 👍🏼

**Review by QuintonJason — APPROVED** [review]
🏆

**Pull request merged**